engine/dlib/src/dlib/log.cpp

```diff
@@ -424,13 +424,49 @@ static android_LogPriority ToAndroidPriority(dmLogSeverity severity)
 }
 #endif
 
+#define MAX_LISTENERS (32)
+static dmLogListener g_dmLog_Listeners[MAX_LISTENERS];
+static int g_dmLog_ListenersCount = 0;
+static bool g_isSendingLogs = false;
+
+void dmLogRegisterListener(dmLogListener listener)
+{
+    if (g_dmLog_ListenersCount >= MAX_LISTENERS) {
+        dmLogWarning("Max dmLog listeners reached (%d)", MAX_LISTENERS);
+    } else {
+        g_dmLog_Listeners[g_dmLog_ListenersCount++] = listener;
+    }
+}
+
+void dmLogUnregisterListener(dmLogListener listener)
+{
+    for (int i = 0; i < g_dmLog_ListenersCount; ++i)
+    {
+        if (g_dmLog_Listeners[i] == listener)
+        {
+            g_dmLog_Listeners[i] = g_dmLog_Listeners[g_dmLog_ListenersCount - 1];
+            g_dmLog_ListenersCount--;
+            return;
+        }
+    }
+    dmLogWarning("dmLog listener not found");
+}
+
+#undef MAX_LISTENERS
+
 void dmLogInternal(dmLogSeverity severity, const char* domain, const char* format, ...)
 {
-    if (!dLib::IsDebugMode())
+    bool is_debug_mode = dLib::IsDebugMode();
+
+    if (!is_debug_mode && g_dmLog_ListenersCount == 0)
```

Comment:
even if it's released, it checks if listeners registered. If so, we do log message string creation